src/d/actor/d_a_shutter.cpp

```diff
@@ -5,80 +5,292 @@
+            cBgD_t* pData = (cBgD_t *)dComIfG_getObjectRes(m_arcname[mType], m_dzbidx[mType]);
+            if (mdBgW[i]->Set(pData, 0x1, &mMtx[i]) == 1) {
+                return FALSE;
+            }
+        }
+        else {
+            return FALSE;
+        }
+    }
+    return TRUE;
 }
 
 /* 000002B8-000004B4       .text Create__11daShutter_cFv */
 s32 daShutter_c::Create() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel[0]->getBaseTRMtx());
+    Vec cullMin = m_cull_min[mType];
+    Vec cullMax = m_cull_max[mType];
+    fopAcM_setCullSizeBox(this, cullMin.x, cullMin.y, cullMin.z, cullMax.x, cullMax.y, cullMax.z);
+    mNo = daShutter_prm::getmNo(this);
+    mcXyz[0].x = -m_width[mType] / 2.0f;
+    mcXyz[1].x = m_width[mType] / 2.0f;
+    if (!fopAcM_isSwitch(this, mNo)) {
+        field_0x328 = 3;
+    }
+    else {
+        field_0x328 = 2;
+        mcXyz[0].x = mcXyz[0].x - m_move_len[mType];
+        mcXyz[1].x = mcXyz[1].x + m_move_len[mType];
+    }
+    mCount = 0x1e;
+    set_mtx();
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        dComIfG_Bgsp()->Regist(mdBgW[i], this);
+        mdBgW[i]->Move();
+    }
+    if (m_open_ev_name[mType] != NULL) {
+        mOpenEventIdx = dComIfGp_evmng_getEventIdx(m_open_ev_name[mType], 0xff);
+    }
+    if (m_close_ev_name[mType] != NULL) {
+        mCloseEventIdx = dComIfGp_evmng_getEventIdx(m_close_ev_name[mType], 0xff);
+    }
+    return 1;
 }
 
 /* 000004B4-000005A0       .text _create__11daShutter_cFv */
 s32 daShutter_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, daShutter_c);
+    mType = daShutter_prm::getType(this);
+    int result = dComIfG_resLoad(&mPhs, m_arcname[mType]);
+    if (result == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, m_heapsize[mType])){
+            return cPhs_ERROR_e;
+        }
+        else {
+            Create();
+        }
+    }
+    return result;
 }
 
 /* 000005E0-000006F0       .text set_mtx__11daShutter_cFv */
 void daShutter_c::set_mtx() {
-    /* Nonmatching */
+    cXyz local_48;
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        mDoMtx_stack_c::YrotS(current.angle.y);
+        mDoMtx_stack_c::multVec(&mcXyz[i], &local_48);
+        mpModel[i]->setBaseScale(scale);
+        mDoMtx_stack_c::transS(current.pos.x + local_48.x, current.pos.y + local_48.y, current.pos.z + local_48.z);
+        mDoMtx_stack_c::ZXYrotM(current.angle.x, current.angle.y, current.angle.z);
+        mpModel[i]->setBaseTRMtx(mDoMtx_stack_c::get());
+        MTXCopy(mDoMtx_stack_c::get(), mMtx[i]);
+    }
 }
 
 /* 000006F0-00000788       .text _execute__11daShutter_cFv */
 bool daShutter_c::_execute() {
-    /* Nonmatching */
+    bool bVar1;
+    int iVar3;
+    if (mCount >= 0) {
+        mCount--;
+    }
+    demo();
+    set_mtx();
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        mdBgW[i]->Move();
+    }
+    mSwitch = fopAcM_isSwitch(this, mNo);
```

Comment:
```suggestion
    mbIsSwitch = fopAcM_isSwitch(this, mSwitchNo);
```
isSwitch doesn't return a switch, it takes a switch and returns a boolean for whether the switch is currently set or not.